### ISSUE TITLE
Calendar display issue with the clock module

### ISSUE THREAD

**Issue by yeheng282** [Alexays/Waybar#3132]
On my Arch Linux system, I am encountering the following issue with the calendar display, as shown in the screenshot below. 
I don't know if this is a bug or not. 

version: 0.10.0
wm: hyprland 0.38.0

![image](https://github.com/Alexays/Waybar/assets/98092393/391291d2-457e-4dac-b96e-2d4dffde4555)
![image_2024-04-10_20-50-04](https://github.com/Alexays/Waybar/assets/98092393/0bc161ec-f261-4524-ae85-63fd342e8f39)

**Comment by yeheng282** [comment]
The problem should be here.
[https://github.com/Alexays/Waybar/blob/e8038ef9f3d85f41821a93fcc6de3310ececab2c/src/modules/clock.cpp#L342](url)

**Comment by ordy** [comment]
Yeah, it looks like other users have reported that same issue as well in #1321

**Comment by Yutsuten** [comment]
I was going to open a new issue when I found this, I'll add the message here.

---

This issue started happening on v0.10.0.

As it wasn't fixed in v0.10.1 and v0.10.2, I feel that this is locale related. My locale is `ja_JP.UTF-8`. Using Sway.

In `clock > calendar > format > weekdays`, I have this setting:

```json
{
    "weekdays": "<u><b>{}</b></u>",
}
```

It renders as:

![waybar-no-fix](https://github.com/Alexays/Waybar/assets/7322925/252b56d0-b2b1-403d-bc41-9c94dd08f1df)

Now I put this workaround in the setting:

```json
{
    "weekdays": "<u><b>日 月 火 水 木 金 土</b></u>",
}
```

The workaround renders as:

![waybar-with-fix](https://github.com/Alexays/Waybar/assets/7322925/b06ef48c-898f-4c25-b0a6-d29ebcf5c259)

This is the desired output. I'd like to use the `{}` syntax, but I'll stick with the workaround until a fix is available.
